Makefile

```diff
@@ -1,59 +1,35 @@
 SHELL ?= /bin/bash
 
-# The Docker registry where images are pushed.
-# Note that if you use an org (like on Quay and DockerHub), you should
-# include that: quay.io/foo
-DOCKER_REGISTRY    ?= brigadecore
+################################################################################
+# Version details                                                              #
+################################################################################
 
-GIT_TAG   = $(shell git describe --tags --always 2>/dev/null)
-VERSION   ?= ${GIT_TAG}
-IMAGE_TAG ?= ${VERSION}
+GIT_VERSION = $(shell git describe --always --abbrev=7 --dirty)
 
-BINS = github-gateway check-run
-IMAGES = brigade-github-app brigade-github-check-run
-
-.PHONY: build
-build: $(BINS)
-
-.PHONY: $(BINS)
-$(BINS): bootstrap
-	go build -o bin/$@ ./cmd/$@
-
-# Cross-compile for Docker+Linux
-.PHONY: build-docker-bins
-build-docker-bins: $(addsuffix -docker-bin,$(BINS))
-
-%-docker-bin: bootstrap
-	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -ldflags '$(LDFLAGS)' -o ./rootfs/$* ./cmd/$*
-
-# To use docker-build, you need to have Docker installed and configured. You should also set
-# DOCKER_REGISTRY to your own personal registry if you are not pushing to the official upstream.
-.PHONY: docker-build
-docker-build: build-docker-bins
-docker-build: $(addsuffix -image,$(IMAGES))
-
-%-image:
-	docker build -f Dockerfile.$* -t $(DOCKER_REGISTRY)/$*:$(IMAGE_TAG) .
-
-# You must be logged into DOCKER_REGISTRY before you can push.
-.PHONY: docker-push
-docker-push: $(addsuffix -push,$(IMAGES))
+################################################################################
+# Docker images we build and publish                                           #
+################################################################################
 
-%-push:
-	docker push $(DOCKER_REGISTRY)/$*:$(IMAGE_TAG)
+ifdef DOCKER_REGISTRY
+	DOCKER_REGISTRY := $(DOCKER_REGISTRY)/
+else
+	DOCKER_REGISTRY := brigadecore/
+endif
 
-.PHONY: lint
-lint:
-	golangci-lint run --config ./golangci.yml
+ifdef VERSION
+	IMMUTABLE_DOCKER_TAG := $(VERSION)
+	MUTABLE_DOCKER_TAG   := latest
+else
+	IMMUTABLE_DOCKER_TAG := $(GIT_VERSION)
+	MUTABLE_DOCKER_TAG   := edge
+endif
 
-.PHONY: test
-test:
-	go test ./pkg/...
+################################################################################
+# Utility targets                                                              #
+################################################################################
 
 .PHONY: redeploy
-redeploy: test
-redeploy: docker-build
-redeploy: docker-push
+redeploy: test push-all-images
```

Comment:
Not really the fault of the re-organization here, but an unintended consequence is, if not explicitly defined via a `default` target, make will grab the first target defined when `make` is invoked (sans any particular target) -- so, `make` results in running `redeploy` here.  Again, we were just 'lucky' that `build` used to be the first target listed...

Perhaps we finally define an explicit default and place prior to the first target? `default: build-all-bins` perhaps?

Reply:
(note: `.DEFAULT_GOAL := build-all-bins # or other preferred default goal(s) here` is an alternative)

Reply:
Nice catch. I always forget that default targets are a thing.

Do you think `test` would be a reasonable default goal? That would be my preference.

Reply:
Sure, `test` sounds great.  👍